src/core/ddsc/tests/psmx.c

```diff
@@ -1264,6 +1326,378 @@ CU_Test (ddsc_psmx, basic)
+            CU_ASSERT_FATAL(result > 0);
+            result = dds_take(reader1, (void**)samples1, sinfo, sample_cnt, sample_cnt);
+            CU_ASSERT_FATAL(result > 0);
+            for (int i = 0; i < result; ++i) {
+              j = cnt + i;
+              if (
+                samples1[i]->a != j ||
+                samples1[i]->b != j + 1 ||
+                samples1[i]->c != j + 2
+              ) {
+                match = false;
+              }
+            }
+            cnt += result;
+          }
+          CU_ASSERT_FATAL(match);
+          CU_ASSERT_FATAL(cnt == sample_cnt);
+        }
+        {
+          PsmxType1* samples2[sample_cnt];
+          dds_sample_info_t sinfo[sample_cnt];
+          const dds_entity_t ws2 = dds_create_waitset(participant);
+          dds_entity_t rcond2 = dds_create_readcondition(reader2, DDS_ANY_STATE);
+          memset(samples2, 0x0, sizeof(samples2));
+          dds_waitset_attach(ws2, rcond2, rcond2);
+
+          int j;
+          int cnt = 0;
+          bool match = true;
+          while ( match && cnt < sample_cnt ) {
+            result = dds_waitset_wait(ws2, NULL, 0, DDS_SECS(10));
+            CU_ASSERT_FATAL(result > 0);
+            result = dds_take(reader2, (void**)samples2, sinfo, sample_cnt, sample_cnt);
+            CU_ASSERT_FATAL(result > 0);
+            for (int i = 0; i < result; ++i) {
+              j = cnt + i;
+              if (
+                samples2[i]->xy.x != j ||
+                samples2[i]->xy.y != (uint8_t)j + 1 ||
+                samples2[i]->z != (uint8_t)j + 2
+              ) {
+                match = false;
+              }
+            }
+            cnt += result;
+          }
+          CU_ASSERT_FATAL(match);
+          CU_ASSERT_FATAL(cnt == sample_cnt);
+        }
+      }
+      dds_delete(domain);
+    }
+    {
+      // Test case using a config without specifying a locator.
+      dds_entity_t domain = -1;
+      {
+        // Create domain
+        struct ddsi_config cfg;
+        ddsi_config_init_default(&cfg);
+        cfg.psmx_instances = ddsrt_malloc(sizeof(*cfg.psmx_instances));
+        cfg.psmx_instances->cfg.name = ddsrt_strdup("iox");
+        cfg.psmx_instances->cfg.library = ddsrt_strdup("psmx_iox");
+        cfg.psmx_instances->cfg.config = ddsrt_strdup("SERVICE_NAME=psmx1;"); // config witout a locator
+        cfg.psmx_instances->cfg.priority.isdefault = 1;
+        cfg.psmx_instances->cfg.priority.value = 1000000;
+        cfg.psmx_instances->next = NULL;
+        domain = dds_create_domain_with_rawconfig(domainId, &cfg);
+        config_psmx_elm_free(cfg.psmx_instances);
+      }
+      CU_ASSERT_FATAL(domain > 0);
+      CU_ASSERT_FATAL(domain_has_psmx_instance_name(domain, "CycloneDDS-IOX-PSMX"));
+      dds_entity_t participant = dds_create_participant(domainId, NULL, NULL);
+      CU_ASSERT_FATAL(participant > 0);
+      {
+        // Check that the second half of the locator is all zeros.
```

Comment:
This assumption is based on details in the `iox` plugin implementation; the value assigned to the locator is the return value of the `get_node_id` call on the PSMX interface that returns a `dds_psmx_node_identifier_t`, which is 16 bytes. This happens to be an 64 bit value with trailing zero's, but could in theory be anything.